### PR TITLE
When connecting to the Ethereum network, handle all errors and clearly communicate back to the user

### PR DESCRIPTION
**Pull request by saturnial**

### CODE REVIEW

File: src/router/router.tsx [hunk 1]
```diff
@@ -37,42 +38,66 @@ class AppRouter extends React.Component<Props, {}> {
...
-        const networkId = parseInt(networkIdString, 10);
-        const accounts = await promisify(web3.eth.getAccounts)();
+        const networkID = parseInt(networkIdString, 10);
 
-        if (!accounts.length) {
-            dispatch(setError(web3Errors.UNABLE_TO_FIND_ACCOUNTS));
-            return;
+        if (_.includes(SUPPORTED_NETWORK_IDS, networkID)) {
+            return networkID;
+        } else {
```
Comment (graemecode) [inline src/router/router.tsx]: You don't need this else, since you already return. Unless you think it reads better.

File: src/router/router.tsx [hunk 2]
```diff
@@ -37,42 +38,66 @@ class AppRouter extends React.Component<Props, {}> {
...
-        dispatch(setNetworkId(networkId));
-
-        if (!_.includes(SUPPORTED_NETWORK_IDS, networkId)) {
-            dispatch(setError(web3Errors.UNABLE_TO_FIND_CONTRACTS));
-            return;
+    async getAccounts(web3: Web3): Promise<string[]> {
+        const accounts = await promisify(web3.eth.getAccounts)();
+        if (accounts.length) {
+            return accounts;
+        } else {
```
Comment (graemecode) [inline src/router/router.tsx]: Same here, re: early return

File: src/router/router.tsx (hunk 1 above)
Comment (saturnial) [reply src/router/router.tsx]: I think it reads better but will defer to the team's consensus cc/ @chrismin @nadavhollander

Comment (nadavhollander) [reply src/router/router.tsx]: Yeah, I think it reads better too 👍

**Pull request merged**